src/librustc_codegen_ssa/back/link.rs

```diff
@@ -1121,7 +1138,9 @@ fn link_args<'a, B: ArchiveBuilder<'a>>(cmd: &mut dyn Linker,
 
     // Try to strip as much out of the generated object by removing unused
     // sections if possible. See more comments in linker.rs
-    if !sess.opts.cg.link_dead_code {
+    //
+    // For Yorick, we don't GC sections so as to keep SIR.
```

Comment:
Huh?

Reply:
This is a linking detail. Because the linker deems our SIR section unreferenced (it can't know we will load the binary from disk at runtime), it will remove it unless we tell it not to.

Reply:
See also `--gc-sections` in the man page for `ld`.

Reply:
We probably need to make clearer that this is about linking? Should we also point readers at the `ld` man page for `--gc-sections`? [I have no idea about this stuff.]

Reply:
There's a couple of places in upstream rust where they talk about some of these details, but I suppose it won't hurt to mention it here.

Reply:
Maybe we just need to point people at where to look upstream then?

Reply:
Turns out the upstream comments I was thinking of were about slightly different linking aspects, so I will just add a comment.

Reply:
OK.